src/VisualStudio/Core/Def/Implementation/TableDataSource/Suppression/VisualStudioSuppressionFixService.cs

```diff
@@ -126,48 +114,16 @@ public bool RemoveSuppressions(bool selectedErrorListEntriesOnly, IVsHierarchy p
             }
         }
 
-        private async Task<ImmutableArray<DiagnosticData>> GetAllBuildDiagnosticsAsync(Func<Project, bool> shouldFixInProject, CancellationToken cancellationToken)
+        private async Task<ImmutableArray<DiagnosticData>> GetAllDiagnosticsAsync(Func<Project, bool> shouldFixInProject, CancellationToken cancellationToken)
         {
             var builder = ImmutableArray.CreateBuilder<DiagnosticData>();
-            var buildDiagnostics = _buildErrorDiagnosticService.GetBuildErrors().Where(d => d.ProjectId != null && d.Severity != DiagnosticSeverity.Hidden);
             var solution = _workspace.CurrentSolution;
-            foreach (var diagnosticsByProject in buildDiagnostics.GroupBy(d => d.ProjectId))
+            foreach (var project in solution.Projects)
             {
-                cancellationToken.ThrowIfCancellationRequested();
-
-                if (diagnosticsByProject.Key == null)
-                {
-                    // Diagnostics with no projectId cannot be suppressed.
-                    continue;
-                }
-
-                var project = solution.GetProject(diagnosticsByProject.Key);
-                if (project != null && shouldFixInProject(project))
+                if (shouldFixInProject(project))
                 {
-                    var diagnosticsByDocument = diagnosticsByProject.GroupBy(d => d.DocumentId);
-                    foreach (var group in diagnosticsByDocument)
-                    {
-                        var documentId = group.Key;
-                        if (documentId == null)
-                        {
-                            // Project diagnostics, just add all of them.
-                            builder.AddRange(group);
-                            continue;
-                        }
-
-                        // For document diagnostics, build does not have the computed text span info.
-                        // So we explicitly calculate the text span from the source text for the diagnostics.
-                        var document = project.GetDocument(documentId);
-                        if (document != null)
-                        {
-                            var tree = await document.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
-                            var text = await tree.GetTextAsync(cancellationToken).ConfigureAwait(false);
-                            foreach (var diagnostic in group)
-                            {
-                                builder.Add(diagnostic.WithCalculatedSpan(text));
-                            }
-                        }
-                    }
+                    var diagnostics = await _diagnosticService.GetDiagnosticsAsync(solution, project.Id, includeSuppressedDiagnostics: true, cancellationToken: cancellationToken).ConfigureAwait(false);
```

Comment:
The flag just decide the final filtering applied to diagnostics [here](http://source.roslyn.io/Microsoft.CodeAnalysis.Features/R/f778d33c69a9fe48.html) - the computation of diagnostics always need to include suppressed diagnostics as the result is cached by diagnostic service and any client can request for suppressed diagnostics (e.g. error list when user wants to see suppressed issues).

However, you are right that this code path doesn't need suppressed diagnostics. I will clean it up in next change, thanks.